applications/api-cat/src/main/java/no/acat/spec/Parser.java

```diff
@@ -0,0 +1,60 @@
+package no.acat.spec;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.v3.oas.models.OpenAPI;
+import no.acat.config.Utils;
+import no.acat.spec.converters.OpenApiV3JsonSpecConverter;
+import no.acat.spec.converters.SwaggerJsonSpecConverter;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class Parser {
+    private static final ObjectMapper mapper = Utils.jsonMapper();
+
+    public static OpenAPI parse(String apiSpec) throws IllegalArgumentException {
+        OpenAPI openAPI;
+        if (OpenApiV3JsonSpecConverter.canConvert(apiSpec)) {
+            openAPI = OpenApiV3JsonSpecConverter.convert(apiSpec);
+        } else if (SwaggerJsonSpecConverter.canConvert(apiSpec)) {
+            openAPI = SwaggerJsonSpecConverter.convert(apiSpec);
+        } else {
+            throw new IllegalArgumentException("Unsupported api spec format");
+        }
+        // try deserialize to verify
+        try {
+            // todo there is something wrong with the converter and openApi model implementation
+            // in some case the serialized version does not unserialize
+            String json = mapper.writeValueAsString(openAPI);
+            mapper.readValue(json, OpenAPI.class);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Error verifying OpenAPI deserialization");
+        }
+
+        return openAPI;
+    }
+
+    public static String getSpecFromUrl(String apiSpecUrlString) throws IllegalArgumentException {
+        URL apiSpecUrl;
+        String apiSpec;
+
+        try {
+            apiSpecUrl = new URL(apiSpecUrlString);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("Invalid api spec url: " + apiSpecUrlString);
+        }
+
```

Comment:
Kunne ikke disse to try-blokkene med fordel vært slått sammen?

Reply:
ja, vi kan, faktisk, hvis vi bruker original melding. litt enklere